File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        if (!root) {
            return {}; // Return an empty vector if the tree is empty.
        }

        vector<TreeNode*> lQ = {root};
        vector<int> lRes;

        while (!lQ.empty()) {
            int levelSize = lQ.size();
            for (int i = 0; i < levelSize; i++) {
                TreeNode* lNode = lQ.front();
                lQ.erase(lQ.begin());

                // Check if the current node is the last one in this level.
                if (i == levelSize - 1) {
                    lRes.push_back(lNode->val);
                }

                if (lNode->left) {
                    lQ.push_back(lNode->left);
                }
                if (lNode->right) {
                    lQ.push_back(lNode->right);
                }
            }
        }

        return lRes;
    }
};

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []

        q = [root]
        curr = []
        final = []

        while q:
            curr = []
            for _ in range(len(q)):
                node = q.pop(0)
                curr.append(node.val)
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
            final.append(curr[-1])
            
        return final